upgrade_downgrade/test_canary_upgrade_downgrade.sh

```diff
@@ -0,0 +1,193 @@
+#!/bin/bash
+
+# Copyright Istio Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+set -o pipefail
+set -x
+
+# TODO: Remove hardcoded versions and pass it from run_upgrade_downgrade_test
+# as command-line arguments like test_upgrade_downgrade.sh
+FROM_VERSION="istio-1.7-alpha.a8c04f92e236676977b3ed58132437d9fafc9aed"
+TO_VERSION="istio-1.8-alpha.f0f221ae1024fb2d4a93b355546f094376fb2e47"
+ISTIOPATH="$GOPATH/src/istio.io"
+
+FROM_DIR="${ISTIOPATH}/tools/upgrade_downgrade/from_dir.g9xSYb/${FROM_VERSION}"
```

Comment:
not sure why these random strings are there?

Reply:
As mentioned in chat, I wanted to focus on getting upgrade downgrade parts. All of this will be passed as command-line parameters from `run_upgrade_downgrade_test.sh`.